### PR TITLE
[FLINK-19244] CsvRowDataDeserializationSchema throws cast exception : Row length mismatch.

### PR DESCRIPTION
**Pull request by yingshin**
## What is the purpose of the change

This pull request fix cast exception when the nested row type is null.
Through csv is not a good format to handle nested types, but it's meaningful to make SerDe  consistent.

## Brief change log

If jsonNode.size equals zero, return a null Row/RowData directly.

## Verifying this change

This change added tests and can be verified as follows:
CsvRowDataSerDeSchemaTest.testSerializeDeserializeNestedTypes
CsvRowDeSerializationSchemaTest.testSerializeDeserializeNestedTypes

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: ( no )
  - The runtime per-record code paths (performance sensitive): ( no )
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no )
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? ( no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/CsvRowDataSerDeSchemaTest.java [hunk 1]
```diff
@@ -331,6 +364,16 @@ private Row testDeserialization(
 			.toExternal(deserializedRow);
 	}
 
+	private void testSerDeConsistency(
+		RowData originalRow,
+		CsvRowDataSerializationSchema.Builder serSchemaBuilder,
+		CsvRowDataDeserializationSchema.Builder deserSchemaBuilder) throws Exception {
```
Comment (wuchong) [inline flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/CsvRowDataSerDeSchemaTest.java:370]: Add one more indent for method arguments.

**Review by wuchong — COMMENTED** [review]
LGTM. 

Could you also create a pull request for release-1.11 branch?

File: flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/CsvRowDataSerDeSchemaTest.java (hunk 1 above)
Comment (yingshin) [reply flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/CsvRowDataSerDeSchemaTest.java]: fixed.
I pull request for release-1.11 branch:  https://github.com/apache/flink/pull/13444

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request merged**